examples/nlp/gpt/run_generation.py

```diff
@@ -0,0 +1,165 @@
+    logger = CustomLoggerWrapper(trainer.loggers)
+
+    ptl_model = load_from_nemo(
+        MegatronGPTModel, cfg.model, trainer, strict=True, restore_path=cfg.pretrained_checkpoint.restore_from_path,
+    )
+
+    with open_dict(cfg):
+        # overwrite the model config with the config from the checkpoint
+        cfg.model.encoder_seq_length = ptl_model.cfg.encoder_seq_length
+
+    # pull values from checkpoint
+    trainer_restore_path = trainer.ckpt_path
+
+    # TODO: log this restore path
+    if trainer_restore_path is not None:
+        custom_trainer_state_dict = retrieve_custom_trainer_state_dict(trainer)
+        consumed_samples = custom_trainer_state_dict["consumed_samples"]
+    else:
+        custom_trainer_state_dict = None
+        consumed_samples = 0
+
+    if os.path.exists(gen_file := os.path.join(cfg.exp_manager.explicit_log_dir, "generations", "generations.jsonl")):
+        js_line = json.loads(subprocess.check_output(["tail", "-1", gen_file]).decode("utf_8"))
+        custom_trainer_state_dict = {"step": js_line["step"], "consumed_samples": js_line["consumed_samples"]}
+        consumed_samples = js_line["consumed_samples"]
+
+    init_distributed(trainer, ptl_model, cfg.model.get("transformer_engine", False))
+    """
+    dp_group = parallel_state.get_data_parallel_group()
+    calc_gbs = cfg.model.generation.rollout_micro_batch_size * dp_group.size()
+    with open_dict(cfg):
+        cfg.model.global_batch_size = calc_gbs
+    with open_dict(ptl_model.cfg):
+        ptl_model.cfg.global_batch_size = calc_gbs
+    if hasattr(ptl_model, "global_batch_size"):
+        ptl_model.global_batch_size = calc_gbs
+    """
+    train_data_cfg = cfg.model.data.train_ds
+
+    if cfg.model.data.get("sample", False):
+        # if it is negative, num_samples is None
+        if cfg.trainer.generation.max_steps < 0:
+            num_samples = None
+        else:
+            num_samples = cfg.trainer.generation.max_steps * cfg.model.global_batch_size
+    else:
+        num_samples = None
+    train_ds = build_sft_dataset(
+        train_data_cfg,
+        ptl_model.tokenizer,
+        num_samples,
+        answer_only_loss=True,
+        is_chat=cfg.model.data.chat,
+        special_tokens=cfg.model.data.chat_prompt_tokens,
+    )
+
+    # eos_id = ptl_model.tokenizer.eos_id
+
+    # collate fn to pad to the max seq length in the batch
+    # collate_fn = partial(
+    #    self_rewarding_custom_collate,
+    #    eos_id=eos_id,
+    #    reset_position_ids=cfg.model.data.get("reset_position_ids", False),
+    #    reset_attention_mask=cfg.model.data.get("reset_attention_mask", False),
+    #    eod_mask_loss=cfg.model.data.get("eod_mask_loss", False),
+    # )
+
+    train_dataloader = build_dataloader(
+        cfg=cfg,
+        dataset=train_ds,
+        consumed_samples=consumed_samples,
+        mbs=cfg.model.micro_batch_size,
+        gbs=cfg.model.global_batch_size,
+        collate_fn=eye,
```

Comment:
Can you use `identity_collate` from `nemo_aligner/data/nlp/builders.py`?

Reply:
Possibly, but this is a really cool shoutout to old school MATLAB users, who would appreciate the easter egg

Reply:
They may get confused though because it's not returning an identity matrix...